bitbots_blackboard/bitbots_blackboard/capsules/costmap_capsule.py

```diff
@@ -0,0 +1,390 @@
+"""
+CostmapCapsule
+^^^^^^^^^^^^^^^^^^
+
+Provides information about the cost of different positions and moves.
+"""
+import math
+from typing import TYPE_CHECKING, Dict, List, Optional, Tuple
+
+if TYPE_CHECKING:
+    from bitbots_blackboard.blackboard import BodyBlackboard
+
+import matplotlib.pyplot as plt
+import numpy as np
+import ros2_numpy
+import tf2_ros as tf2
+from bitbots_utils.utils import (get_parameter_dict,
+                                 get_parameters_from_other_node)
+from geometry_msgs.msg import (Point, Pose, PoseStamped,
+                               PoseWithCovarianceStamped, Quaternion,
+                               TransformStamped, TwistStamped,
+                               TwistWithCovarianceStamped)
+from nav_msgs.msg import MapMetaData, OccupancyGrid
+from PIL import Image, ImageDraw
+from rclpy.clock import ClockType
+from rclpy.duration import Duration
+from rclpy.time import Time
+from scipy.interpolate import griddata
+from scipy.ndimage import gaussian_filter
+from soccer_vision_3d_msgs.msg import RobotArray, Robot
+from std_msgs.msg import Header
+from std_srvs.srv import Trigger
+from tf2_geometry_msgs import PointStamped
+from tf_transformations import euler_from_quaternion, quaternion_from_euler
+
+class CostmapCapsule:
+    def __init__(self, blackboard: "BodyBlackboard"):
+        self.blackboard = blackboard
+        self.body_config = get_parameter_dict(self._blackboard.node, "body")
+
+
+        self.tf_buffer = tf2.Buffer(cache_time=Duration(seconds=30))
+        self.tf_listener = tf2.TransformListener(self.tf_buffer, self._blackboard.node)
+
+        self.base_footprint_frame: str = self._blackboard.node.get_parameter('base_footprint_frame').value
+
+        parameters = get_parameters_from_other_node(
```

Comment:
It is also fine to get the field dimensions from the param blackboard in both capsules like this.